### PR TITLE
Update marketing spec, and add README about how to do it

### PR DESCRIPTION
**Pull request by Moishe**
### Description
The previous spec file was edited to make it possible to create SDKs. I backfilled most of the edits into the code that's used to serve the spec file in production. This PR contains that new spec file, as well as a README about how to manually keep it up-to-date.

Don’t bother diffing the spec files themselves because the formatting is different and everything gets weird (unless you want to download the files manually and pipe ’em through `json.tool` or what-have-you). Future updates to the spec file should be more readable though, since they’ll be coming from a consistent source.

### CODE REVIEW

File: spec/README.md
Comment (prichey) [inline spec/README.md:1]: Good call throwing a README in here!

**Comment by prichey** [comment]
Adding @mtashley who should probably be in the loop about changes here

**Comment by Moishe** [comment]
Also here's a sample SDK diff (newly-generated SDK unzipped from artifacts and copied to a fork of the SDK): https://github.com/mailchimp/mailchimp-marketing-python/pull/2

**Review by prichey — APPROVED** [review]
Everything here LGTM! Thanks for helping streamline the update process

**Comment by Moishe** [comment]
@mtashley I'll wait until tomorrow to merge just to make sure you've had a chance to see this.

**Review by mtashley — APPROVED** [review]
Excellent

**Pull request merged**
